### DISCUSSION TITLE
Channel stuck in stopping mode

### DISCUSSION THREAD

**Discussion by rutledj** [nextgenhealthcare/connect#6097] [Q&A]
I have a channel reader that is stuck in Stopping mode. Halt will not stop it. What causes this and how can it be fixed?  We had an issue with the password on the destination connecting to sftp but corrected that. 

![image](https://github.com/nextgenhealthcare/connect/assets/6484700/b692b92c-a4b8-4fb1-b62d-ad9cc35a1ef9)

**Comment by jonbartels** [comment]
You said the channel is a Channel Reader - When another channel sends a message to it the source connector thread runs in the destination thread for the channel writer or `VMRouter` call. If you stop the channel that writes to this channel does the Source state change?

If practical, take a thread dump from Java to find out what it is stuck on. [Heres a HOWTO](https://www.digitalocean.com/community/tutorials/java-thread-dump-visualvm-jstack-kill-3-jcmd) I like the `kill -3` method. This won't help you kill it but it will show where its stuck. If its waiting on data or a DB or something you can kill the underlying connection and free it up.

**Comment by rutledj** [comment]
Sorry. My mistake. It is a file reader.  Can any of the thread dump be accomplished with MC running on appliance (when you don't have access to the vm?

**Reply by jonbartels** [reply]
Yes. There is an undocumented tool called "expert pack" that will take a thread dump. `https://yourappliance/expert` use the same login as you use for the Control Panel.
